### PR TITLE
Fixtimesinfinity

### PR DESCRIPTION
**Pull request by mmatera**
This PR tries to fix errors and inconsistencies in expressions involving products of complex numbers and Infinity quantities.

### CODE REVIEW

File: mathics/builtin/arithmetic.py
Comment (rocky) [inline mathics/builtin/arithmetic.py:1101]: flake8 correctly informs me that "DirectedInfinity[0]" appears twice. See below in line 1116. The second occurrence (that is, not this one), will be the one used. 

If that is what is _desired_ then it would be better to remove this line.

**Review by rocky — APPROVED** [review]
LGTM - but please note the comment about the duplicate dictionary key that appears.

**Comment by mmatera** [comment]
Thanks! I fixed this here. Shall I merge this if this pass the tests?

**Comment by rocky** [comment]
> Thanks! I fixed this here. Shall I merge this if this pass the tests?

Yes - and then if we could _rebase_ (not merge) master onto the extensecmptests branch that would be good.

When we have several branches that all want to go into master with the exact code, it is hard to understand what is going on individually because of the duplication which obscures the essential part of the PR. 

Thanks.

**Comment by mmatera** [comment]
Indeed. I am trying to keep the extensecmptests updated with the changes in fixtimesinfinity, and make that more tests pass. What I wasn't sure is if it is better to merge the parts that are working now, or to have a merge with the complete fix.

**Comment by rocky** [comment]
Ok - then let's merge this. I'll do the rebase after the merge.

**Pull request merged**
